src/infill/SpaghettiInfill.cpp

```diff
@@ -0,0 +1,139 @@
+/** Copyright (C) 2017 Ultimaker - Released under terms of the AGPLv3 License */
+#include "SpaghettiInfill.h"
+
+namespace cura {
+
+
+void SpaghettiInfill::generateSpaghettiInfill(SliceMeshStorage& mesh)
+{
+    coord_t layer_height = mesh.getSettingInMicrons("layer_height");
+    coord_t line_width = mesh.getSettingInMicrons("infill_line_width");
+    double layer_height_mm = INT2MM(layer_height);
+    int spaghetti_max_layer_count = std::max(1, static_cast<int>(mesh.getSettingInMicrons("spaghetti_max_height") / layer_height));
+    // TODO: account for the initial layer height
+
+    coord_t filling_area_inset = mesh.getSettingInMicrons("spaghetti_inset");
+    
+    if (mesh.getSettingInAngleDegrees("spaghetti_max_infill_angle") >= 90)
+    {
+        return; // infill cannot be combined into pillars
```

Comment:
This is something that should be limited from the GUI as well. We now allow an angle of 90 in the GUI.